### PR TITLE
Adds support for Xcode Cloud as CI environment

### PR DESCRIPTION
**Pull request by rpassis**
///////////////////⚡///////////////////

This PR adds basic Xcode Cloud CI source support to `danger`

For reference, the documentation for all available environment variables defined by a build running on Xcode Cloud can be found [here](https://developer.apple.com/documentation/xcode/environment-variable-reference)

///////////////////⚡///////////////////

### CODE REVIEW

File: lib/danger/ci_source/xcode_cloud.rb [hunk 1]
```diff
@@ -0,0 +1,40 @@
+require "danger/request_sources/github/github"
+
```
Comment (rpassis) [inline lib/danger/ci_source/xcode_cloud.rb:2]:
```suggestion
```

File: spec/lib/danger/ci_sources/ci_source_spec.rb [hunk 2]
```diff
@@ -5,35 +5,36 @@
     it "returns list of CI subclasses" do
       expect(described_class.available_ci_sources.map(&:to_s)).to match_array(
         [
-          "Danger::LocalGitRepo",
```
Comment (rpassis) [inline spec/lib/danger/ci_sources/ci_source_spec.rb:8]: I've added `Danger::XcodeCloud` as a source here, and then sorted them alphabetically for better organization.

**Comment by orta** [comment]
Perfect, this looks great!

**Comment by orta** [comment]
Thanks, this is shipped in 8.4.0!

**Pull request merged**